Source/WebCore/css/CSSSelector.cpp

```diff
@@ -1040,7 +1045,13 @@ void CSSSelector::replaceNestingParentByPseudoClassScope()
 bool CSSSelector::hasExplicitNestingParent() const
 {
     auto checkForExplicitParent = [] (const CSSSelector& selector) {
-        if (selector.match() == CSSSelector::Match::NestingParent)
+        if (selector.match() == CSSSelector::Match::UnknownToSerialize) {
+            auto tokens = CSSTokenizer(selector.value()).tokenRange();
```

Comment:
It’s not OK to use a token range after the tokenizer has been destroyed. I suggest writing this:
```
    CSSTokenizer tokenizer(selector.value());
    for (auto& token : tokenizer.tokenRange())
```

Reply:
Good catch indeed, the token range is non owning. The `-asan` build is correctly catching this 👍